File: Taewoo/BOJ10866.cpp
#include<iostream>
#include<queue>
using namespace std;

int main() {
    deque<int> q;

    int N; cin>>N;

    for(int i=0; i<N; i++) {
        string input; cin>>input;

        if(input == "push_front") {
            int x; cin>>x;
            q.push_front(x);
        }
        else if(input == "push_back") {
            int x; cin>>x;
            q.push_back(x);
        }
        else if(input == "pop_front") {
            if(q.size()) {
                cout<<q.front()<<'\n'; q.pop_front();
            }
            else cout<<-1<<'\n';
        }
        else if(input == "pop_back") {
            if(q.size()) {
                cout<<q.back()<<'\n'; q.pop_back();
            }
            else cout<<-1<<'\n';
        }
        else if(input == "size") cout<<q.size()<<'\n';
        else if(input == "empty") {
            if(!q.empty()) cout<<0<<'\n';
            else cout<<1<<'\n';
        }
        else if(input == "front") {
            if(q.size()) cout<<q.front()<<'\n';
            else cout<<-1<<'\n';
        }
        else if(input == "back") {
            if(q.size()) cout<<q.back()<<'\n';
            else cout<<-1<<'\n';
        }
    }
    return 0;
}

File: SaeHyeon/Algorithms/Data_structure/BOJ10866.py
from collections import deque
import sys

input=sys.stdin.readline

def main():
    n=int(input())
    queue=deque()
    for i in range(n):
        s=input().split()
        if s[0] == 'push_front':
            queue.appendleft(int(s[1]))
        if s[0] == 'push_back':
            queue.append(int(s[1]))
        if s[0] == 'pop_front':
            if not queue:
                print(-1)
            else:
                print(queue.popleft())
        if s[0] == 'pop_back':
            if not queue:
                print(-1)
            else:
                print(queue.pop())
        if s[0] == 'size':
            print(len(queue))
        if s[0] == 'empty':
            if not queue:
                print(1)
            else:
                print(0)
        if s[0] == 'front':
            if not queue:
                print(-1)
            else:
                print(queue[0])
        if s[0] == 'back':
            if not queue:
                print(-1)
            else:
                print(queue[-1])


if __name__ == "__main__":
    main()